drivers/flash/flash_stm32_qspi.c

```diff
@@ -32,9 +30,59 @@ LOG_MODULE_REGISTER(flash_stm32_qspi, CONFIG_FLASH_LOG_LEVEL);
 #define STM32_QSPI_FIFO_THRESHOLD         8
 #define STM32_QSPI_CLOCK_PRESCALER_MAX  255
 
-#define STM32_QSPI_USE_DMA DT_NODE_HAS_PROP(DT_PARENT(DT_DRV_INST(0)), dmas)
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_qspi_nor) \
+	|| DT_HAS_COMPAT_STATUS_OKAY(st_stm32_qspi_nor_ospi)
 
-#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_qspi_nor)
+#ifdef CONFIG_FLASH_STM32_OSPI_MIMIC_QSPI
+	/* Mimic QSPI using OCTOSPI if QUADSPI
+	 * is not available
+	 */
+#define DT_DRV_COMPAT st_stm32_qspi_nor_ospi
+
+/* SPI Specific functions and variables */
+#define HAL_QSPI_TYPEDEF				OCTOSPI_TypeDef
+#define HAL_QSPI_HANDLETYPEDEF			OSPI_HandleTypeDef
+#define HAL_QSPI_COMMAND				OSPI_RegularCmdTypeDef
+#define HAL_QSPI_SEND_COMMAND_IT		HAL_OSPI_Command_IT
+#define HAL_QSPI_RECEIVE_DMA			HAL_OSPI_Receive_DMA
+#define HAL_QSPI_TRANSMIT_DMA			HAL_OSPI_Transmit_DMA
+#define HAL_QSPI_RECEIVE_IT				HAL_OSPI_Receive_IT
+#define HAL_QSPI_TRANSMIT_OT			HAL_OSPI_Transmit_IT
```

Comment:
Is this called _OT on purpose, or is this a typo?

Reply:
Fixed.